File: test_streamable_client.py
#!/usr/bin/env python
"""
Test client for Streamable HTTP MCP server
"""

import asyncio
import json
from mcp.client.session import ClientSession
from mcp.client.streamable_http import StreamableHTTPTransport

async def main():
    # Streamable HTTP MCP server URL - no /sse suffix needed!
    url = "http://localhost:8001/mcp"
    
    print(f"Connecting to Streamable HTTP MCP server: {url}")
    
    try:
        # Create Streamable HTTP transport
        transport = StreamableHTTPTransport(url)
        
        # Connect and get streams
        async with transport.connect() as (read_stream, write_stream):
            # Create MCP client session with the streams
            client = ClientSession(read_stream, write_stream)
            
            # Initialize the session
            await client.initialize()
            print("✅ Connected to MCP server!")
            
            # List available tools
            print("\n📋 Available tools:")
            tools_result = await client.list_tools()
            for tool in tools_result.tools:
                print(f"  - {tool.name}: {tool.description}")
            
            print("\n🔧 Testing tools:")
            
            # Test echo tool
            print("\n1. Testing 'echo' tool...")
            result = await client.call_tool("echo", {"msg": "Hello Streamable HTTP!"})
            print(f"   Result: {result.content[0].text}")
            
            # Test add tool
            print("\n2. Testing 'add' tool...")
            result = await client.call_tool("add", {"x": 25, "y": 17})
            print(f"   Result: {result.content[0].text}")
            
            # Test multiply tool
            print("\n3. Testing 'multiply' tool...")
            result = await client.call_tool("multiply", {"x": 4.5, "y": 3.0})
            print(f"   Result: {result.content[0].text}")
            
            # Test say_hello tool
            print("\n4. Testing 'say_hello' tool...")
            result = await client.call_tool("say_hello", {})
            print(f"   Result: {result.content[0].text}")
            
            print("\n🎉 All Streamable HTTP tests completed successfully!")
        
    except Exception as e:
        print(f"❌ Error connecting to MCP server: {e}")
        print(f"   Error type: {type(e).__name__}")
        import traceback
        traceback.print_exc()
        print("\nTroubleshooting:")
        print("1. Make sure the server is running on localhost:8001")
        print("2. Check that the server supports Streamable HTTP at /mcp")
        print("3. Verify the server logs for any errors")

if __name__ == "__main__":
    asyncio.run(main())

File: test_streamable_client.js
#!/usr/bin/env node

/**
 * Test client for Streamable HTTP MCP server using JavaScript SDK
 * This mirrors exactly what OAP does
 */

const { StreamableHTTPClientTransport } = require("@modelcontextprotocol/sdk/client/streamableHttp.js");
const { Client } = require("@modelcontextprotocol/sdk/client/index.js");

async function main() {
  // Streamable HTTP MCP server URL - matches what OAP expects
  const url = "http://localhost:8001/mcp";
  
  console.log(`Connecting to Streamable HTTP MCP server: ${url}`);
  
  try {
    // Create transport exactly like OAP does
    const connectionClient = new StreamableHTTPClientTransport(new URL(url));
    const mcp = new Client({
      name: "test-client",
      version: "1.0.0",
    });

    // Connect like OAP does
    await mcp.connect(connectionClient);
    console.log("✅ Connected to MCP server!");
    
    // List available tools
    console.log("\n📋 Available tools:");
    const tools = await mcp.listTools();
    for (const tool of tools.tools) {
      console.log(`  - ${tool.name}: ${tool.description}`);
    }
    
    console.log("\n🔧 Testing tools:");
    
    // Test echo tool
    console.log("\n1. Testing 'echo' tool...");
    const result1 = await mcp.callTool({
      name: "echo",
      arguments: { msg: "Hello Streamable HTTP from JS!" }
    });
    console.log(`   Result: ${result1.content[0].text}`);
    
    // Test add tool
    console.log("\n2. Testing 'add' tool...");
    const result2 = await mcp.callTool({
      name: "add",
      arguments: { x: 25, y: 17 }
    });
    console.log(`   Result: ${result2.content[0].text}`);
    
    // Test multiply tool
    console.log("\n3. Testing 'multiply' tool...");
    const result3 = await mcp.callTool({
      name: "multiply", 
      arguments: { x: 4.5, y: 3.0 }
    });
    console.log(`   Result: ${result3.content[0].text}`);
    
    // Test say_hello tool
    console.log("\n4. Testing 'say_hello' tool...");
    const result4 = await mcp.callTool({
      name: "say_hello",
      arguments: {}
    });
    console.log(`   Result: ${result4.content[0].text}`);
    
    console.log("\n🎉 All Streamable HTTP tests completed successfully!");
    
    // Close the connection
    await mcp.close();
    
  } catch (error) {
    console.log(`❌ Error connecting to MCP server: ${error.message}`);
    console.log(`   Error type: ${error.constructor.name}`);
    console.error(error);
    console.log("\nTroubleshooting:");
    console.log("1. Make sure the server is running on localhost:8001");
    console.log("2. Check that the server supports Streamable HTTP at /mcp");
    console.log("3. Verify the server logs for any errors");
    console.log("4. Make sure you have @modelcontextprotocol/sdk installed");
  }
}

if (require.main === module) {
  main();
}